From cea226f19df8cb78086374698d52f30485c2df43 Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Thu, 11 Aug 2022 10:35:15 +0530
Subject: [PATCH] wpa_supplicant : add wpa_cli support for cac

support to display cac inprogress state is added
as part of 'wpa_cli -i wlanx status' command

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 wpa_supplicant/ctrl_iface.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/wpa_supplicant/ctrl_iface.c b/wpa_supplicant/ctrl_iface.c
index 61aeea7..b96a996 100644
--- a/wpa_supplicant/ctrl_iface.c
+++ b/wpa_supplicant/ctrl_iface.c
@@ -2205,7 +2205,7 @@ static int wpa_supplicant_ctrl_iface_status(struct wpa_supplicant *wpa_s,
 					    const char *params,
 					    char *buf, size_t buflen)
 {
-	char *pos, *end, tmp[30];
+	char *pos, *end, tmp[30], *tmp_buf;
 	int res, verbose, wps, ret;
 #ifdef CONFIG_HS20
 	const u8 *hs20;
@@ -2346,12 +2346,27 @@ static int wpa_supplicant_ctrl_iface_status(struct wpa_supplicant *wpa_s,
 	}
 #endif /* CONFIG_SAE */
 #endif /* CONFIG_SME */
+	if (wpa_s->ifmsh) {
+		if(wpa_s->wpa_state == WPA_SCANNING && wpa_s->ifmsh->cac_started)
+			tmp_buf = "inprogress";
+		else
+			tmp_buf = "N/A";
+	} else {
+		tmp_buf = "N/A";
+	}
 	ret = os_snprintf(pos, end - pos, "wpa_state=%s\n",
 			  wpa_supplicant_state_txt(wpa_s->wpa_state));
 	if (os_snprintf_error(end - pos, ret))
 		return pos - buf;
 	pos += ret;
 
+	ret = os_snprintf(pos, end - pos, "cac=%s\n",
+			  tmp_buf);
+	if (os_snprintf_error(end - pos, ret))
+		return pos - buf;
+	pos += ret;
+
+
 	if (wpa_s->l2 &&
 	    l2_packet_get_ip_addr(wpa_s->l2, tmp, sizeof(tmp)) >= 0) {
 		ret = os_snprintf(pos, end - pos, "ip_address=%s\n", tmp);
-- 
2.7.4

